static/app/views/discover/eventDetails/content.tsx

```diff
@@ -155,159 +157,179 @@ class EventDetailsContent extends DeprecatedAsyncComponent<Props, State> {
-                        const childTransactionLink = eventDetailsRoute({
-                          eventSlug: childTransactionProps.eventSlug,
-                          orgSlug: organization.slug,
-                        });
-
-                        return {
-                          pathname: childTransactionLink,
-                          query: eventView.generateQueryStringObject(),
-                        };
-                      },
-                    }}
-                  >
-                    <QuickTraceContext.Provider value={results}>
-                      {hasProfilingFeature ? (
-                        <ProfilesProvider
-                          orgSlug={organization.slug}
-                          projectSlug={this.projectId}
-                          profileId={profileId || ''}
-                        >
-                          <ProfileContext.Consumer>
-                            {profiles => (
-                              <ProfileGroupProvider
-                                type="flamechart"
-                                input={
-                                  profiles?.type === 'resolved' ? profiles.data : null
-                                }
-                                traceID={profileId || ''}
-                              >
-                                <BorderlessEventEntries
-                                  organization={organization}
-                                  event={event}
-                                  project={projects[0] as Project}
-                                  location={location}
-                                  showTagSummary={false}
-                                />
-                              </ProfileGroupProvider>
-                            )}
-                          </ProfileContext.Consumer>
-                        </ProfilesProvider>
-                      ) : (
-                        <BorderlessEventEntries
-                          organization={organization}
-                          event={event}
-                          project={projects[0] as Project}
-                          location={location}
-                          showTagSummary={false}
-                        />
-                      )}
-                    </QuickTraceContext.Provider>
-                  </SpanEntryContext.Provider>
-                ) : (
-                  <LoadingIndicator />
-                )
-              }
-            </Projects>
-          </Layout.Main>
-          {isSidebarVisible && (
-            <Layout.Side>
-              <EventVitals event={event} />
-              <EventCustomPerformanceMetrics
+    ) => {
+      if (metaResults && isTransaction(event)) {
+        const {isLoading} = metaResults;
+
+        if (isLoading) {
+          return <LoadingIndicator />;
+        }
+
+        handleTraceDetailsRouting(
+          metaResults,
+          event,
+          organization,
+          location
+        );
```

Comment:
This feels kind of fishy to add in the render method and more like a side effect. Typically side effects are handled in componentDidUpdate. Can we try moving it there?

Reply:
I think this might be responsible for some console errors, so hopefully moving it to `componentDidUpdate` fixes them

Reply:
Yeah, like we discussed, pushed the logic to the useEffect of a wrapping component.